server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponse.java

```diff
@@ -171,45 +171,33 @@ public record ShardView(
         String node,
         boolean nodeIsDesired,
         @Nullable String relocatingNode,
-        boolean relocatingNodeIsDesired,
+        @Nullable Boolean relocatingNodeIsDesired,
         int shardId,
         String index,
         @Nullable Double forecastWriteLoad,
         @Nullable Long forecastShardSizeInBytes
     ) implements Writeable, ToXContentObject {
 
         private static final TransportVersion ADD_FORECASTS_VERSION = TransportVersion.V_8_7_0;
+        private static final TransportVersion NULLABLE_RELOCATING_NODE_IS_DESIRED = TransportVersion.V_8_8_0;
 
         public static ShardView from(StreamInput in) throws IOException {
-            if (in.getTransportVersion().onOrAfter(ADD_FORECASTS_VERSION)) {
-                return new ShardView(
-                    ShardRoutingState.fromValue(in.readByte()),
-                    in.readBoolean(),
-                    in.readOptionalString(),
-                    in.readBoolean(),
-                    in.readOptionalString(),
-                    in.readBoolean(),
-                    in.readVInt(),
-                    in.readString(),
-                    in.readOptionalDouble(),
-                    in.readOptionalLong()
-                );
-            } else {
-                var shardView = new ShardView(
-                    ShardRoutingState.fromValue(in.readByte()),
-                    in.readBoolean(),
-                    in.readOptionalString(),
-                    in.readBoolean(),
-                    in.readOptionalString(),
-                    in.readBoolean(),
-                    in.readVInt(),
-                    in.readString(),
-                    null,
-                    null
-                );
+            var shardView = new ShardView(
+                ShardRoutingState.fromValue(in.readByte()),
+                in.readBoolean(),
+                in.readOptionalString(),
+                in.readBoolean(),
+                in.readOptionalString(),
+                in.readOptionalBoolean(), // not checking NULLABLE_RELOCATING_NODE_IS_DESIRED as readOptionalBoolean also reads readBoolean
```

Comment:
True, but ehh that's a bit too tricksy for my liking. Let's check the version here so that the reading code aligns properly with the writing code.